nautobot/core/api/schema.py

```diff
@@ -0,0 +1,34 @@
+import re
+
+from drf_spectacular.openapi import AutoSchema
+
+
+class NautobotAutoSchema(AutoSchema):
+    """Nautobot-specific extensions to drf-spectacular's AutoSchema."""
+
+    def get_operation_id(self):
+        """Extend the base method to handle Nautobot's REST API bulk operations.
+
+        Without this extension, every one of our ModelViewSet classes will result in drf-spectacular complaining
+        about operationId collisions, e.g. between DELETE /api/dcim/devices/ and DELETE /api/dcim/devices/<pk>/ would
+        both get resolved to the same "dcim_devices_destroy" operation-id and this would make drf-spectacular complain.
+
+        With this extension, the bulk endpoints automatically get a different operation-id from the non-bulk endpoints.
+        """
+        if hasattr(self.view, "action") and self.view.action in ["bulk_update", "bulk_partial_update", "bulk_destroy"]:
```

Comment:
It seems like it might be better to assign this list of actions as a class attribute so it doesn't get re-defined and evaluated every time this method is called?